test/e2e/util/options.go

```diff
@@ -0,0 +1,431 @@
+package util
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	hypershiftaws "github.com/openshift/hypershift/cmd/cluster/aws"
+	"github.com/openshift/hypershift/cmd/cluster/azure"
+	"github.com/openshift/hypershift/cmd/cluster/core"
+	"github.com/openshift/hypershift/cmd/cluster/kubevirt"
+	"github.com/openshift/hypershift/cmd/cluster/none"
+	hypershiftopenstack "github.com/openshift/hypershift/cmd/cluster/openstack"
+	"github.com/openshift/hypershift/cmd/cluster/powervs"
+	awscmdutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	azurenodepool "github.com/openshift/hypershift/cmd/nodepool/azure"
+	kubevirtnodepool "github.com/openshift/hypershift/cmd/nodepool/kubevirt"
+	openstacknodepool "github.com/openshift/hypershift/cmd/nodepool/openstack"
+	"github.com/openshift/hypershift/cmd/version"
+	controlplaneoperatoroverrides "github.com/openshift/hypershift/hypershift-operator/controlplaneoperator-overrides"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Options are global test options applicable to all scenarios.
+type Options struct {
+	LatestReleaseImage   string
+	PreviousReleaseImage string
+	N2MinorReleaseImage  string
+	N1MinorReleaseImage  string
+	IsRunningInCI        bool
+	ArtifactDir          string
+
+	// BeforeApply is a function passed to the CLI create command giving the test
+	// code an opportunity to inspect or mutate the resources the CLI will create
+	// before they're applied.
+	BeforeApply func(client.Object) `json:"-"`
+
+	Platform    hyperv1.PlatformType
+	PlatformRaw string
+
+	ConfigurableClusterOptions ConfigurableClusterOptions
+	AdditionalTags             stringSliceVar
+
+	IssuerURL                string
+	ServiceAccountSigningKey []byte
+
+	// If set, the CreateCluster test will create a cluster with request serving
+	// isolation topology.
+	RequestServingIsolation bool
+
+	// If testing request serving isolation topology, we need a kubeconfig to the
+	// parent of the management cluster, name and namespace of the management cluster
+	// so we can create additional nodepools for it.
+	ManagementParentKubeconfig string
+	ManagementClusterNamespace string
+	ManagementClusterName      string
+	// If set, the UpgradeControlPlane test will upgrade control plane without
+	// reconciling PKI.
+	DisablePKIReconciliation bool
+
+	HyperShiftCLIPath              string
```

Comment:
this and `OCPath` are not needed anymore, right?

Reply:
correct, good catch! fixed!